global_planner_name: "moveit_hybrid_planning/MoveItPlanningPipeline"

# The rest of these parameters are typical for moveit_cpp
planning_scene_monitor_options:
  name: "planning_scene_monitor"
  robot_description: "robot_description"
  joint_state_topic: "/lbr/joint_states"
  # attached_collision_object_topic: "/moveit_cpp/planning_scene_monitor"
  attached_collision_object_topic: "/lbr/attached_collision_object"

  # Subscribe to this topic (The name comes from the perspective of moveit_cpp)
  publish_planning_scene_topic: "/lbr/planning_scene"
  # Publish this topic, e.g. to visualize with RViz
  # monitored_planning_scene_topic: "/global_planner/planning_scene"
  monitored_planning_scene_topic: "/lbr/monitored_planning_scene"

  wait_for_initial_state_timeout: 10.0

plan_request_params:
  planner_id: "RRTStarkConfigDefault"
  planning_pipeline: ompl
  # planner_id: "LIN"
  # planning_pipeline: pilz_industrial_motion_planner
  planning_attempts: 5
  planning_time: 5.0
  max_velocity_scaling_factor: 1.0
  max_acceleration_scaling_factor: 1.0
  cartesian_limits:
    x_min: -100.0
    x_max: 100.0
    y_min: -100.0
    y_max: 100.0
    z_min: -100.0
    z_max: 100.0

planning_pipelines:
  #namespace: "moveit_cpp"  # optional, default is ~
  pipeline_names: ["ompl", "pilz_industrial_motion_planner"]
  # planning_pipelines:
  # - ompl
  # - pilz_industrial_motion_planner
  # planning_plugin: ompl_interface/OMPLPlanner
  # pipeline_names: ["ompl"]
  namespace: "lbr"

# ompl ns should be outside of planning_pipelines ns
ompl:
  arm:
    planner_configs:
    - SBLkConfigDefault
    - ESTkConfigDefault
    - LBKPIECEkConfigDefault
    - BKPIECEkConfigDefault
    - KPIECEkConfigDefault
    - RRTkConfigDefault
    - RRTConnectkConfigDefault
    - RRTstarkConfigDefault
    - TRRTkConfigDefault
    - PRMkConfigDefault
    - PRMstarkConfigDefault
    - FMTkConfigDefault
    - BFMTkConfigDefault
    - PDSTkConfigDefault
    - STRIDEkConfigDefault
    - BiTRRTkConfigDefault
    - LBTRRTkConfigDefault
    - BiESTkConfigDefault
    - ProjESTkConfigDefault
    - LazyPRMkConfigDefault
    - LazyPRMstarkConfigDefault
    - SPARSkConfigDefault
    - SPARStwokConfigDefault
    - TrajOptDefault
    enforce_constrained_state_space: true
  planner_configs:
    BFMTkConfigDefault:
      balanced: 0
      cache_cc: 1
      extended_fmt: 1
      heuristics: 1
      nearest_k: 1
      num_samples: 1000
      optimality: 1
      radius_multiplier: 1.0
      type: geometric::BFMT
    BKPIECEkConfigDefault:
      border_fraction: 0.9
      failed_expansion_score_factor: 0.5
      min_valid_path_fraction: 0.5
      range: 0.0
      type: geometric::BKPIECE
    BiESTkConfigDefault:
      range: 0.0
      type: geometric::BiEST
    BiTRRTkConfigDefault:
      cost_threshold: 1.0e+300
      frountier_node_ratio: 0.1
      frountier_threshold: 0.0
      init_temperature: 100
      range: 0.0
      temp_change_factor: 0.1
      type: geometric::BiTRRT
    ESTkConfigDefault:
      goal_bias: 0.05
      range: 0.0
      type: geometric::EST
    FMTkConfigDefault:
      cache_cc: 1
      extended_fmt: 1
      heuristics: 0
      nearest_k: 1
      num_samples: 1000
      radius_multiplier: 1.1
      type: geometric::FMT
    KPIECEkConfigDefault:
      border_fraction: 0.9
      failed_expansion_score_factor: 0.5
      goal_bias: 0.05
      min_valid_path_fraction: 0.5
      range: 0.0
      type: geometric::KPIECE
    LBKPIECEkConfigDefault:
      border_fraction: 0.9
      min_valid_path_fraction: 0.5
      range: 0.0
      type: geometric::LBKPIECE
    LBTRRTkConfigDefault:
      epsilon: 0.4
      goal_bias: 0.05
      range: 0.0
      type: geometric::LBTRRT
    LazyPRMkConfigDefault:
      range: 0.0
      type: geometric::LazyPRM
    LazyPRMstarkConfigDefault:
      type: geometric::LazyPRMstar
    PDSTkConfigDefault:
      type: geometric::PDST
    PRMkConfigDefault:
      max_nearest_neighbors: 10
      type: geometric::PRM
    PRMstarkConfigDefault:
      type: geometric::PRMstar
    ProjESTkConfigDefault:
      goal_bias: 0.05
      range: 0.0
      type: geometric::ProjEST
    RRTConnectkConfigDefault:
      range: 0.0
      type: geometric::RRTConnect
      # bounds:
      #   cartesian_limits: [-1, 1, -1, 1, -1, 1]
    RRTkConfigDefault:
      goal_bias: 0.05
      range: 0.0
      type: geometric::RRT
    RRTstarkConfigDefault:
      delay_collision_checking: 1
      goal_bias: 0.1
      range: 0.0
      type: geometric::RRTstar
      # bounds:
      #   cartesian_limits: [-1, 1, -1, 1, 0, 0.4]
    SBLkConfigDefault:
      range: 0.0
      type: geometric::SBL
    SPARSkConfigDefault:
      dense_delta_fraction: 0.001
      max_failures: 1000
      sparse_delta_fraction: 0.25
      stretch_factor: 3.0
      type: geometric::SPARS
    SPARStwokConfigDefault:
      dense_delta_fraction: 0.001
      max_failures: 5000
      sparse_delta_fraction: 0.25
      stretch_factor: 3.0
      type: geometric::SPARStwo
    STRIDEkConfigDefault:
      degree: 16
      estimated_dimension: 0.0
      goal_bias: 0.05
      max_degree: 18
      max_pts_per_leaf: 6
      min_degree: 12
      min_valid_path_fraction: 0.2
      range: 0.0
      type: geometric::STRIDE
      use_projected_distance: 0
    TRRTkConfigDefault:
      frountierNodeRatio: 0.1
      frountier_threshold: 0.0
      goal_bias: 0.05
      init_temperature: 1.0e-05
      k_constant: 0.0
      max_states_failed: 10
      min_temperature: 1.0e-09
      range: 0.0
      temp_change_factor: 2.0
      type: geometric::TRRT
    TrajOptDefault:
      type: geometric::TrajOpt
  # one is sufficient
  planning_plugin: ompl_interface/OMPLPlanner
  planning_plugins:
  - ompl_interface/OMPLPlanner
  # can be either a vector of strings or a single string
  request_adapters: >-
    default_planner_request_adapters/AddTimeOptimalParameterization
    default_planner_request_adapters/ResolveConstraintFrames 
    default_planner_request_adapters/FixWorkspaceBounds
    default_planner_request_adapters/FixStartStateBounds 
  # default_planner_request_adapters/FixStartStateCollision
  # default_planner_request_adapters/FixStartStatePathConstraints

  # can be either a vector of strings or a single string
  response_adapters: >-
    default_planning_response_adapters/AddTimeOptimalParameterization
    default_planning_response_adapters/ValidateSolution
    default_planning_response_adapters/DisplayMotionPath
  start_state_max_bounds_error: 0.1

pilz_industrial_motion_planner:
  capabilities: >-
    pilz_industrial_motion_planner/MoveGroupSequenceAction
    pilz_industrial_motion_planner/MoveGroupSequenceService
  default_planner_config: LIN
  planning_plugin: pilz_industrial_motion_planner/CommandPlanner
  planning_plugins:
  - pilz_industrial_motion_planner/CommandPlanner
  request_adapters: >-
    default_planner_request_adapters/ResolveConstraintFrames
    default_planner_request_adapters/FixWorkspaceBounds 
    default_planner_request_adapters/FixStartStateBounds
    default_planner_request_adapters/FixStartStateCollision 
    default_planner_request_adapters/FixStartStatePathConstraints
# part of pilz_industrial_motion_planner
robot_description_planning:
  cartesian_limits:
    max_rot_vel: 1.57
    max_trans_acc: 2.25
    max_trans_dec: -5.0
    max_trans_vel: 1.0

moveit_controller_manager: moveit_simple_controller_manager/MoveItSimpleControllerManager
moveit_manage_controllers: true
moveit_simple_controller_manager:
  controller_names:
  - joint_trajectory_controller
  joint_trajectory_controller:
    action_ns: follow_joint_trajectory
    default: true
    joints:
    - A1
    - A2
    - A3
    - A4
    - A5
    - A6
    - A7
    type: FollowJointTrajectory

# move_group params, can be deleted
# publish_geometry_updates: true
# publish_planning_scene: true
# capabilities: ''
# disable_capabilities: ''
# publish_robot_description: false
# publish_robot_description_semantic: true
# publish_state_updates: true
# allow_trajectory_execution: true
# monitor_dynamics: false
# publish_transforms_updates: true

# can be deleted
# arm:
#   kinematics_solver: null
# default_planning_pipeline: ompl
# planning_plugin: ompl_interface/OMPLPlanner

robot_description_kinematics:
  arm:
    kinematics_solver: pick_ik/PickIkPlugin
    kinematics_solver_timeout: 0.05
    kinematics_solver_attempts: 3
    mode: global
    position_scale: 1.0
    rotation_scale: 0.5
    position_threshold: 0.001
    orientation_threshold: 0.01
    cost_threshold: 0.001
    minimal_displacement_weight: 0.0
    gd_step_size: 0.0001

  default_acceleration_scaling_factor: 0.1
  default_velocity_scaling_factor: 0.1
  joint_limits:
    A1:
      angle_wraparound: false
      has_acceleration_limits: true
      has_deceleration_limits: false
      has_effort_limits: false
      # has_jerk_limits: null
      has_position_limits: false
      has_soft_limits: false
      has_velocity_limits: true
      k_position: .nan
      k_velocity: .nan
      max_acceleration: 10.0
      max_effort: .nan
      max_jerk: .nan
      # max_position: null
      max_velocity: 1.710422666954443
      # min_position: null
      min_velocity: .nan
      soft_lower_limit: .nan
      soft_upper_limit: .nan
    A2:
      angle_wraparound: false
      has_acceleration_limits: true
      has_deceleration_limits: false
      has_effort_limits: false
      # has_jerk_limits: null
      has_position_limits: false
      has_soft_limits: false
      has_velocity_limits: true
      k_position: .nan
      k_velocity: .nan
      max_acceleration: 10.0
      max_effort: .nan
      max_jerk: .nan
      # max_position: null
      max_velocity: 1.710422666954443
      # min_position: null
      min_velocity: .nan
      soft_lower_limit: .nan
      soft_upper_limit: .nan
    A3:
      angle_wraparound: false
      has_acceleration_limits: true
      has_deceleration_limits: false
      has_effort_limits: false
      # has_jerk_limits: null
      has_position_limits: false
      has_soft_limits: false
      has_velocity_limits: true
      k_position: .nan
      k_velocity: .nan
      max_acceleration: 10.0
      max_effort: .nan
      max_jerk: .nan
      # max_position: null
      max_velocity: 1.7453292519943295
      # min_position: null
      min_velocity: .nan
      soft_lower_limit: .nan
      soft_upper_limit: .nan
    A4:
      angle_wraparound: false
      has_acceleration_limits: true
      has_deceleration_limits: false
      has_effort_limits: false
      # has_jerk_limits: null
      has_position_limits: false
      has_soft_limits: false
      has_velocity_limits: true
      k_position: .nan
      k_velocity: .nan
      max_acceleration: 10.0
      max_effort: .nan
      max_jerk: .nan
      # max_position: null
      max_velocity: 2.2689280275926285
      # min_position: null
      min_velocity: .nan
      soft_lower_limit: .nan
      soft_upper_limit: .nan
    A5:
      angle_wraparound: false
      has_acceleration_limits: true
      has_deceleration_limits: false
      has_effort_limits: false
      # has_jerk_limits: null
      has_position_limits: false
      has_soft_limits: false
      has_velocity_limits: true
      k_position: .nan
      k_velocity: .nan
      max_acceleration: 10.0
      max_effort: .nan
      max_jerk: .nan
      # max_position: null
      max_velocity: 2.443460952792061
      # min_position: null
      min_velocity: .nan
      soft_lower_limit: .nan
      soft_upper_limit: .nan
    A6:
      angle_wraparound: false
      has_acceleration_limits: true
      has_deceleration_limits: false
      has_effort_limits: false
      # has_jerk_limits: null
      has_position_limits: false
      has_soft_limits: false
      has_velocity_limits: true
      k_position: .nan
      k_velocity: .nan
      max_acceleration: 10.0
      max_effort: .nan
      max_jerk: .nan
      # max_position: null
      max_velocity: 3.141592653589793
      # min_position: null
      min_velocity: .nan
      soft_lower_limit: .nan
      soft_upper_limit: .nan
    A7:
      angle_wraparound: false
      has_acceleration_limits: true
      has_deceleration_limits: false
      has_effort_limits: false
      # has_jerk_limits: null
      has_position_limits: false
      has_soft_limits: false
      has_velocity_limits: true
      k_position: .nan
      k_velocity: .nan
      max_acceleration: 10.0
      max_effort: .nan
      max_jerk: .nan
      # max_position: null
      max_velocity: 3.141592653589793
      # min_position: null
      min_velocity: .nan
      soft_lower_limit: .nan
      soft_upper_limit: .nan